contracts/identity/lib.rs

```diff
@@ -1,14 +1,51 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
+use ink::storage::traits::StorageLayout;
+
+/// Each identity will be associated with a unique identifier called `IdentityNo`.
+pub type IdentityNo = u64;
+
+/// We want to keep the address type very generic since we want to support any
+/// address format. We won't actually keep the addresses in the contract itself.
+/// Before storing them, we'll encrypt them to ensure privacy.
+pub type Address = Vec<u8>;
+
+/// Used to represent any blockchain in the Polkadot, Kusama or Rococo network.
+#[derive(scale::Encode, scale::Decode, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo, StorageLayout))]
+pub enum Network {
+	/// Polkadot network
+	Polkadot(String),
+	/// Kusama network
+	Kusama(String),
+	/// Rococo network
+	Rococo(String),
+}
+
+#[derive(scale::Encode, scale::Decode, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo, StorageLayout))]
+pub struct IdentityInfo {
+	/// Each address is associated with a specific blockchain.
+	addresses: Vec<(Network, Address)>,
```

Comment:
I would prefer use of Mapping for time efficiency

Reply:
Yes, that is what I was thinking also, but it is not possible: https://substrate.stackexchange.com/questions/1659/how-to-have-a-mapping-in-a-custom-structure-inside-an-ink-contract

Reply:
Is it possible to declare a storage item like `mapping(IdentityNo, mapping(Network, Address))?

Reply:
No, I have tried.

Reply:
okay, got it.
seems like we have no other choice than using Vec.

Reply:
Yes, that just seems like the most reasonable choice we have.